tools/parabricks/parabricks_fq2bam.xml

```diff
@@ -0,0 +1,145 @@
+<tool id="parabricks_fq2bam" name="Parabricks fq2bam" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="@PROFILE@">
+    <description>Run GPU-bwa mem mapper</description>
+    <macros>
+        <token name="@TOOL_VERSION@">4.4.0-1</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+        <token name="@PROFILE@">24.2</token>
+    </macros>
+    <creator>
+        <organization name="European Galaxy Team" url="https://galaxyproject.org/eu/"/>
+        <person givenName="Anup" familyName="Kumar" email="kumara@informatik.uni-freiburg.de"/>
+    </creator>
+    <requirements>
+        <container type="docker">nvcr.io/nvidia/clara/clara-parabricks:@TOOL_VERSION@</container>
+    </requirements>
+    <command detect_errors="aggressive">
+    <![CDATA[
+        #if $reference_source.reference_source_selector == 'cached':
+            ln -f -s '$reference_source.ref_file.fields.path' reference.fasta &&
+        #end if
```

Comment:
I'm surprised this should work.

What `$reference_source.ref_file.fields.path` will evaluate to is the `.fa` file in a cached directory, which is similar to your test-data directory here.
But by symlinking that file into the job working directory under a different name, you're, effectively, leaving the index file behind and that's why the bwa mem wrapper uses `$reference_source.ref_file.fields.path` directly without symlinking.

Does fq2bam follow the symlink and use the real filename, or is it indexing the file again?

Reply:
`
ln -f -s '/home/ubuntu/data/galaxy_codebase/galaxytools/nvidia_parabricks/galaxytools/tools/parabricks/tool-data/bwa-mem-mt-genome.fa' reference.fasta && ln -f -s '/tmp/tmpfm6q80dz/files/b/a/e/dataset_bae26d7a-ff80-4065-bbef-2482fa341170.dat' input1.fastq && pbrun fq2bam --low-memory --ref reference.fasta --standalone-bqsr --in-se-fq input1.fastq --out-bam 'output_mapping.bam'
`

so, it works from the `tool-data` folder and local index files are accessible

Reply:
Directory structure of .fa and other supporting index files (for simulating situation 2)

<img width="228" height="220" alt="dripwd" src="https://github.com/user-attachments/assets/c70a5fb3-655d-45b4-8bc8-006abc3ba1cb" />